packages/expect-utils/src/jasmineUtils.ts

```diff
@@ -115,7 +115,7 @@ function eq(
       // Coerce dates to numeric primitive values. Dates are compared by their
       // millisecond representations. Note that invalid dates with millisecond representations
       // of `NaN` are not equivalent.
-      return +a == +b;
+      return decodeURI(a) == decodeURI(b);
```

Comment:
Instead using decodeURI
we could just write
```if(isNaN(a)) {
    if(isNaN(b)) { 
   return false;
  }
 }

and let the rest be same

Reply:
Looks good too

Reply:
`return (isNaN(a) && isNaN(b)) || +a == +b;`

Reply:
Can you think any scenario that is returning true using `+a == +b`?